### PR TITLE
Add well position validation for sample rules

### PR DESCRIPTION
**Pull request by eliottBo**
## Description
This PR addresses part of [#3020](https://github.com/Clinical-Genomics/cg/issues/3020). It is part of improve-order-flow-main.
### Added

- It adds a validation for the well position format for orders without cases.

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/services/order_validation_service/errors/sample_errors.py [hunk 1]
```diff
@@ -52,3 +52,8 @@ class OrganismDoesNotExistError(SampleError):
 class SampleNameNotAvailableError(SampleError):
     field: str = "name"
     message: str = "Sample name already used in previous order"
+
+
+class WellFormatError(SampleError):
+    field: str = "well_position"
+    message: str = "Well position is not in correct format"
```
Comment (islean) [inline cg/services/order_validation_service/errors/sample_errors.py:59]: Would be nice to give the customers knowledge of how it should look like. Maybe "Well position must follow the format 'A-H:1-12'"?

File: cg/services/order_validation_service/rules/sample/rules.py [hunk 2]
```diff
@@ -121,3 +123,15 @@ def validate_sample_names_available(
...
     return errors
+
+
+def validate_sample_well_position_format(
+    order: OrderWithNonHumanSamples, **kwargs
+) -> list[WellFormatError]:
+    errors: list[WellFormatError] = []
+    for sample_index, sample in order.enumerated_samples:
+        if sample.is_on_plate:
+            if not is_valid_well_format(sample.well_position):
```
Comment (islean) [inline cg/services/order_validation_service/rules/sample/rules.py:134]: Mostly a flavour thing - you could consider baking the `sample.is_on_plate`check in to a `does_sample_have_an_invalid_well_format` or something to hide away more of the logic and get rid of an indentation. I am fine with keeping it as is as well.

**Review by islean — APPROVED** [review]
Super nice!

File: cg/services/order_validation_service/rules/sample/rules.py (hunk 2 above)
Comment (eliottBo) [reply cg/services/order_validation_service/rules/sample/rules.py]: I changed accordingly lmk if it is what you expected :)

Comment (islean) [reply cg/services/order_validation_service/rules/sample/rules.py]: looks great!

File: cg/services/order_validation_service/rules/sample/utils.py [hunk 3]
```diff
@@ -60,3 +62,11 @@ def get_indices_for_repeated_sample_names(order: OrderWithNonHumanSamples) -> li
         if counter.get(sample.name) > 1:
             indices.append(index)
     return indices
+
+
+def does_sample_have_an_invalid_well_format(sample: Sample) -> bool:
```
Comment (ChrOertlin) [inline cg/services/order_validation_service/rules/sample/utils.py:67]:
```suggestion
def is_valid_well_format(sample: Sample) -> bool:
```

**Review by ChrOertlin — APPROVED** [review]
Looks good!

File: cg/services/order_validation_service/rules/sample/utils.py (hunk 3 above)
Comment (eliottBo) [reply cg/services/order_validation_service/rules/sample/utils.py:67]: Thanks! would be `is_invalid_well_format` in this case I will update!

Comment (islean) [reply cg/services/order_validation_service/rules/sample/utils.py]: I mean, for me the `is_valid_well_format` implies that we are only evaluating the format of the well position, but it actually determines if the samples well format is relevant as well. Very minor thing but I would probably call it what you had and if needed, break out the matching with the regex into a function with that name

**Pull request merged**